src/main/scala/generate/database/DatabaseGenerator.scala

```diff
@@ -0,0 +1,7 @@
+package generate.database
+
+import generate.database.ast.Statement
+
+trait DatabaseGenerator {
+  def generate(statement: Statement): String
```

Comment:
Also on the method, so classes that implement this should get the comments for free